lambda-runtime/src/main/scala/feral/lambda/runtime/LambdaRuntimeEnv.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.runtime
+
+import cats.MonadThrow
+import cats.effect.std.Env
+import cats.syntax.all._
+import org.http4s.Uri
+
+trait LambdaRuntimeEnv[F[_]] {
+  def lambdaFunctionName: F[String]
+  def lambdaFunctionMemorySize: F[Int]
+  def lambdaFunctionVersion: F[String]
+  def lambdaLogGroupName: F[String]
+  def lambdaLogStreamName: F[String]
+  def lambdaRuntimeApi: F[Uri]
+}
```

Comment:
Do we know if these values ever change while the process is running? If not maybe we should load them from the `Env` once, and keep them cached.

I cannot imagine why they'd change, but I know Lambda does rely on other envs changing at runtime, so it wouldn't be surprising.

Reply:
Ah, this might have been a mistake on my end. The [custom lambda runtime docs](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html) lists reading environment variables as an initialization task while the runtime is currently reading them for each invocation. So no, I don't think they will change so I think it's safe to read them once at the initialization step instead.

Reply:
Ah this might have been my mistake. The [custom lambda runtime docs](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html) mention that reading environment variables is an initialization task while the custom runtime is currently reading them for every invocation. So no, I don't think they will change so I think it is safe to read them once during initialization.

Reply:
Thanks! No not at all, I think it was my bad suggestion in https://github.com/typelevel/feral/pull/276#discussion_r982731850 😅 sorry!

So then let's refactor this so they are read once during initialization.

Reply:
Yes, please go ahead!